src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AlleleFractionLikelihoods.java

```diff
@@ -0,0 +1,179 @@
+        return hetLogLikelihood(state, segment, count, indicator, AllelicPanelOfNormals.EMPTY_PON);
+    }
+
+    public static double hetLogLikelihood(final AlleleFractionState state, final int segment, final AllelicCount count, final AlleleFractionIndicator indicator,
+                                          final AllelicPanelOfNormals allelicPON) {
+        final SimpleInterval site = count.getInterval();
+        final double alpha;
+        final double beta;
+        if (allelicPON.equals(AllelicPanelOfNormals.EMPTY_PON)) {
+            beta = state.meanBias() / state.biasVariance();
+            alpha = state.meanBias() * beta;
+
+        } else {
+            alpha = allelicPON.getAlpha(site);
+            beta = allelicPON.getBeta(site);
+        }
+        final double pi = state.outlierProbability();
+        final double minorFraction = state.minorFractionInSegment(segment);
+        final int a = count.getAltReadCount();
+        final int r = count.getRefReadCount();
+
+        return hetLogLikelihood(alpha, beta, minorFraction, pi, indicator, a, r);
+    }
+
+    public static double hetLogLikelihood(final double alpha, final double beta, final double minorFraction,
+                                          final double outlierProbability, final AlleleFractionIndicator indicator,
+                                          final int a, final int r) {
+        if (indicator == AlleleFractionIndicator.OUTLIER) {
+            return log(outlierProbability) + log10ToLog(log10Factorial(a) + log10Factorial(r) - log10Factorial(a + r + 1));
+        } else {
+            final double f = indicator == AlleleFractionIndicator.ALT_MINOR ? minorFraction : 1 - minorFraction;
+            return log((1 - outlierProbability) / 2) + logPhi(alpha, beta, f, a, r);
+        }
+    }
+
+    protected static double logPhi(final double alpha, final double beta, final double f, final int a, final int r) {
+        final double lambda0 = lambda0(alpha, beta, f, a, r);
+        final int n = a + r;
+        final double kappa = kappa(alpha, f, r, n, lambda0);
+        final double rho = rho(lambda0, kappa);
+        final double tau = tau(lambda0, kappa);
+        final double logc = alpha*log(beta) - Gamma.logGamma(alpha) + a * log(f) + r * log(1 - f)
+                + (r + alpha - rho) * log(lambda0) + (tau - beta) * lambda0 - n * log(f + (1 - f) * lambda0);
+        return logc + Gamma.logGamma(rho) - rho * log(tau);
+    }
+
+    protected static double lambda0(final double alpha, final double beta, final double f, final int a, final int r) {
+        final double w = (1 - f) * (a - alpha + 1) + beta * f;
+        return (sqrt(w * w + 4 * beta * f * (1 - f) * (r + alpha - 1)) - w) / (2 * beta * (1 - f));
+    }
+
+    protected static double kappa(final double alpha, final double f, final int r, final int n, final double lambda0) {
```

Comment:
Similarly, `kappa` is the curvature of the posterior evaluate at lambda0 -- we're using the mode and curvature at the mode to define a gamma approximation to integrate analytically.